account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py

```diff
@@ -0,0 +1,357 @@
+                    company.provision_pl_analytic_account_id.id or
+                    False)
+
+                line_ids = self._create_move_and_lines(
+                    amount, company.provision_bs_gain_account_id.id,
+                    company.provision_pl_gain_account_id.id, sums, label,
+                    form, partner_id, currency_id,
+                    analytic_credit_acc_id=analytic_acc_id)
+
+                created_ids.extend(line_ids)
+
+        # under revaluation
+        elif amount <= -0.01:
+            amount = -amount
+            reval_loss_account = company.revaluation_loss_account_id
+            if reval_loss_account:
+
+                analytic_acc_id = (company.revaluation_analytic_account_id.id
+                                   if company.revaluation_analytic_account_id
+                                   else False)
+
+                line_ids = self._create_move_and_lines(
+                    amount, reval_loss_account.id, account_id, sums, label,
+                    form, partner_id, currency_id,
+                    analytic_debit_acc_id=analytic_acc_id)
+
+                created_ids.extend(line_ids)
+
+            if company.provision_bs_loss_account_id and \
+               company.provision_pl_loss_account_id:
+
+                analytic_acc_id = (
+                    company.provision_pl_analytic_account_id and
+                    company.provision_pl_analytic_account_id.id or
+                    False)
+
+                line_ids = self._create_move_and_lines(
+                    amount, company.provision_pl_loss_account_id.id,
+                    company.provision_bs_loss_account_id.id, sums, label,
+                    form, partner_id, currency_id,
+                    analytic_debit_acc_id=analytic_acc_id)
+
+                created_ids.extend(line_ids)
+
+        return created_ids
+
+    @staticmethod
+    def _check_company(company):
+        return (not company.revaluation_loss_account_id and
+                not company.revaluation_gain_account_id and
+                not (company.provision_bs_loss_account_id and
+                     company.provision_pl_loss_account_id) and
+                not (company.provision_bs_gain_account_id and
+                     company.provision_pl_gain_account_id))
+
+    @api.multi
+    def revaluate_currency(self):
+        """
+        Compute unrealized currency gain and loss and add entries to
+        adjust balances
+
+        @return: dict to open an Entries view filtered on generated move lines
+        """
+
+        account_obj = self.env['account.account']
+
+        company = self.journal_id.company_id or self.env.user.company_id
+        if self._check_company(company):
+            raise UserError(
+                _("No revaluation or provision account are defined"
+                  " for your company.\n"
+                  "You must specify at least one provision account or"
+                  " a couple of provision account.")
```

Comment:
I would add :
"You must specify at least one provision account or a couple of provision account **in the accounting settings**"